src/main/java/com/revature/rpm/components/Listener.java

```diff
@@ -0,0 +1,102 @@
+package com.revature.rpm.components;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.jboss.logging.Logger;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClient;
+import com.amazonaws.services.sqs.model.DeleteMessageRequest;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.revature.rpm.entities.Notification;
+import com.revature.rpm.services.NotificationService;
+
+@Component
+public class Listener implements InitializingBean {
+
+	// Injecting environment variable data into strings
+	@Value("${MESSAGING_ACCESS_KEY}")
+	private String accessKey;
+	
+	@Value("${MESSAGING_SECRET_ACCESS_KEY}")
+	private String secretKey;
+	
+	@Value("${MESSAGING_REGION}")
+	private String region;
+	
+	@Value("${MESSAGING_QUEUE_URL}")
+	private String queueUrl;
+	
+	// ObjectMapper can map between Java/JSON
+	@Autowired
+	ObjectMapper objectMapper;
+	
+	@Autowired
+	NotificationService notificationService;
+	
+	Logger logger = Logger.getLogger(Listener.class);
+
+	private BasicAWSCredentials credentials;
+	private AmazonSQS sqsClient;
+	
+	// afterPropertiesSet is the appropriate place to initialize them
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		credentials = new BasicAWSCredentials(accessKey, secretKey);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).

Reply:
We are using temp credentials, this snippet constructs an access object after reading environment variables with values assigned through IAM access keys/secret keys.

Reply:
In addition, I assigned an IAM user and a temporary access key/secret combination to each of my group members instead of using just one access key.